docs/proxy-keys/proxy-key-sols.tex

```diff
@@ -0,0 +1,319 @@
+Following is a high-level view of the technical architecture for both the Ledger and Wallet based solutions.
+Ultimately our goal is to deliver on the functional and non-functional requirements, without adding
+unnecessary complexity to the system components. Critically, we will also call out specific benefits \&
+drawbacks, which should inform the approach choice.
+
+\subsection{A Ledger based approach}
+Conceptually, the Ledger based solution to deliver Proxy-keys is predicated on an extension of the existing
+delegation mechanism. To begin, let us review how delegation is currently executed in the Ledger layer. 
+Currently, users can formally stake their ADA balance to a given stake pool by posting a \emph{delegation certificate} 
+to the blockchain.
+
+\begin{description}
+  \item A delegation certificate is a 2-tuple containing:
+  \begin{itemize}
+    \item the stake address delegating its stake rights, $\mathcal{A}_{s,\text{source}}$.
+    \item the stake pool (verification key hash) to which stake is delegated, $\mathcal{H}(vk_\text{pool})$.
+  \end{itemize}
+  Posting a delegation certificate requires a signature from the private key counterpart of the delegating 
+  stake address $\mathcal{S}(\mathcal{A}_{s,\text{source}})$.
+\end{description} 
+
+Next we propose an extension to this mechanism to satisfy the Proxy-key requirements.
+
+\pagebreak
+
+\subsubsection{High-level design}
+Chiefly, the main enhancement to the existing delegation mechanism is the proposal of a \emph{Proxy delegation certificate}.
+
+\begin{description}
+  \item A Proxy delegation certificate is a 3-tuple containing:
+  \begin{itemize}
+    \item the stake address proxying its stake rights, $\mathcal{A}_{s,\text{source}}$, i.e. the \gls{trustor}.
+    \item the stake address to which rights are being conferred, $\mathcal{A}_{p,\text{target}}$, i.e. the \gls{trustee}.
+    \item the amount to be conferred upon the Proxy-key, $\mathcal{N}_{p,\text{index}}$.
+  \end{itemize}
+  Posting a proxy delegation certificate requires a signature from the private key counterpart of the source 
+  stake address $\mathcal{S}(\mathcal{A}_{s,\text{source}})$.
+\end{description} 
+
+With this approach, we effectively create a "pointer" to the regular delegation certificate, which can be used to specify to the Ledger:
+\begin{itemize}
+  \item the identity of the \gls{trustee}.
+  \item the amount the \gls{trustee} is empowered to stake.
+  \item the identity of the \gls{trustor}.
+  \item the approval of the \gls{trustor}.
+\end{itemize}
+
+This mechanism will allow us to meet most of the requirements described in the Proxy-key product requirements document,
+however, it does have some limitations and indeed some complexities to be considered.
+
+\subsubsection{Benefits \& drawbacks}
+Next we will explore the benefits and drawbacks of the Ledger based solution described above. We will also explore
+which requirements described in the PRD would be possible to meet.
```

Comment:
Maybe this is just me, but what is a PRD? product requirements document?

Reply:
yep exactly that.